File: insertion_sort.py
def insertion_sort(A):
    # For every element after the first,
    for j in range(1, len(A)):
        key = A[j]
        i = j-1
        # Look for greater previous elements,
        while i >= 0 and A[i] > key:
            # Scoot them forward,
            A[i+1] = A[i]
            i -= 1
        # And put the element before them.
        A[i+1] = key
    return A

File: InsertionSort.java
package Standard_Algorithms;

public class InsertionSort
{
    public static void insertionSort(int[] A)
    {
        for (int j=1; j<A.length; j++)
            {
                int key = A[j];
                int i = j-1;
                while ((i >= 0) && (A[i] > key))
                    {
                        A[i+1] = A[i];
                        i--;
                    }
                A[i+1] = key;
            }
    }

}